tests/tests_discard_passphrase.yml

```diff
@@ -0,0 +1,45 @@
+---
+- hosts: all
+
+  vars:
+    nbde_client_bindings:
+      - device: "{{ nbde_client_test_device }}"
+        pass: "{{ nbde_client_test_pass }}"
+        discard_passphrase: yes
+        servers:
+          - http://localhost
+      - device: "{{ nbde_client_test_device }}"
+        servers:
+          - http://localhost
+
+  tasks:
+    - name: Set up test environment
+      include: tasks/setup_test.yml
+
+    - name: Run the test
+      block:
+        - name: Use nbde_client role
+          include_role:
+            name: linux-system-roles.nbde_client
+
+        - name: Attempt to unlock device
+          include: tasks/verify_unlock_device.yml
+
+        - name: Make sure the attempt to unlock succeeded
+          assert:
+            that:
+              - not nbde_client_unlock.failed
+              - not nbde_client_close.failed
+
+        - name: Attempt to check whether default passphrase works
+          include: tasks/verify_default_passphrase.yml
+
+        - name: Make sure the default passphrase did not work
+          assert:
+            that:
+              - nbde_client_passphrase.rc != 0
```

Comment:
since `nbde_client_passphrase` is a register variable, use `- nbde_client_passphrase is success` rather than checking `.rc` directly - see https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#task-results